dataproc/create_cluster_and_submit_job.py

```diff
@@ -0,0 +1,223 @@
+    already been uploaded to `bucket_name`"""
+    job_details = {
+        "projectId": project,
+        "job": {
+            "placement": {
+                "clusterName": cluster_name
+            },
+
+            "pysparkJob": {
+                "mainPythonFileUri": 'gs://{}/{}'.format(bucket_name, FILENAME)
+            }
+        }
+    }
+    result = dataproc.projects().regions().jobs().submit(
+        projectId=project,
+        region=REGION,
+        body=job_details).execute()
+    jobId = result['reference']['jobId']
+    print("Submitted job ID {}".format(jobId))
+    return jobId
+# [END submit_pyspark_job]
+
+
+# [START delete]
+def delete_cluster(dataproc, project, cluster):
+    print "Tearing down cluster"
+    result = dataproc.projects().regions().clusters().delete(
+        projectId=project,
+        region=REGION,
+        clusterName=cluster).execute()
+    return result
+# [END delete]
+
+
+# [START wait]
+def wait_for_job(dataproc, project, job_id):
+    print('Waiting for job to finish...')
+    while True:
+        result = dataproc.projects().regions().jobs().get(
+            projectId=project,
+            region=REGION,
+            jobId=job_id).execute()
+        # Handle exceptions
+        if result['status']['state'] == 'ERROR':
+            raise Exception(result['error'])
+        elif result['status']['state'] == 'DONE':
+            print("Job finished")
+            return result
+# [END wait]
+
+
+# [START get_client]
+def get_client():
+    """Builds an http client authenticated with the service account
+    credentials."""
+    credentials = GoogleCredentials.get_application_default()
+    dataproc = discovery.build('dataproc', 'v1', credentials=credentials)
+    return dataproc
+# [END get_client]
+
+
+def create_and_submit_main(project_id, zone, cluster_name, bucket_name):
+    dataproc = get_client()
+    try:
+        create_cluster(dataproc, project_id, cluster_name, zone)
+        wait_for_cluster_creation(dataproc, project_id, cluster_name, zone)
+        upload_pyspark_file(project_id, bucket_name)
+        cluster_list = list_clusters_with_details(
+            dataproc, project_id)['clusters']
+
+        (cluster_id, output_bucket) = (
+            get_cluster_id_by_name(cluster_list, cluster_name))
+        job_id = submit_pyspark_job(
+            dataproc, project_id, cluster_name, bucket_name)
+        wait_for_job(dataproc, project_id, job_id)
+
+        output = download_output(project_id, cluster_id, output_bucket, job_id)
+        print("Received job output {} ".format(output))
+        return output
+    finally:
+        delete_cluster(dataproc, project_id, cluster_name)
+
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        '--project_id', help='Project ID you want to access.', required=True),
```

Comment:
I think the names make it a lot easier to parse the run command, and it looks needlessly complicated to make positional arguments that can also be named. Though happy to see a snippet that makes it simpler